File: content/problems/3039-apply-operations-to-make-string-empty/solutions/maximum-frequency-last-occurrences/Solution.java
class Solution {
    public String lastNonEmptyString(String s){
        int[] frequency=new int[26],lastIndex=new int[26];
        for(int i=0;i<s.length();i++){
            int letter=s.charAt(i)-'a';
            frequency[letter]++;
            lastIndex[letter]=i;
        }
        int maximum=0;
        for(int count:frequency)maximum=Math.max(maximum,count);
        StringBuilder answer=new StringBuilder();
        for(int i=0;i<s.length();i++){
            int letter=s.charAt(i)-'a';
            if(frequency[letter]==maximum&&lastIndex[letter]==i)answer.append(s.charAt(i));
        }
        return answer.toString();
    }
}

File: content/problems/3039-apply-operations-to-make-string-empty/solutions/maximum-frequency-last-occurrences/solution.cpp
class Solution {
    public:string lastNonEmptyString(string s){
        array<int,26> frequency{
        },lastIndex{
        };
        for(int i=0;i<(int)s.size();++i){
            int letter=s[i]-'a';
            ++frequency[letter];
            lastIndex[letter]=i;
        }
        int maximum=*max_element(frequency.begin(),frequency.end());
        string answer;
        for(int i=0;i<(int)s.size();++i){
            int letter=s[i]-'a';
            if(frequency[letter]==maximum&&lastIndex[letter]==i)answer.push_back(s[i]);
        }
        return answer;
    }
};
